src/composables/useGetParticipants.js

```diff
@@ -34,30 +31,56 @@ export function useGetParticipants(isActive = ref(true), isTopBar = true) {
 	 */
 	function initialiseGetParticipants() {
 		EventBus.on('joined-conversation', onJoinedConversation)
-
+		EventBus.on('signaling-users-in-room', updateUsersFromInternalSignaling)
+		EventBus.on('signaling-users-joined', updateUsersJoinedFromExternalSignaling)
+		EventBus.on('signaling-users-left', updateUsersLeftFromExternalSignaling)
+		EventBus.on('signaling-users-changed', updateUsersChangedFromExternalSignaling)
+		EventBus.on('signaling-all-users-changed-in-call-to-disconnected', updateUsersCallDisconnectedFromExternalSignaling)
 		// FIXME this works only temporary until signaling is fixed to be only on the calls
 		// Then we have to search for another solution. Maybe the room list which we update
 		// periodically gets a hash of all online sessions?
 		EventBus.on('signaling-participant-list-changed', debounceUpdateParticipants)
 		subscribe('guest-promoted', onJoinedConversation)
 	}
 
+	const updateUsersFromInternalSignaling = async ([participants]) => {
+		const hasUnknownSessions = await store.dispatch('updateParticipantsFromInternalSignaling', { token: token.value, participants })
+		if (hasUnknownSessions) {
+			debounceUpdateParticipants()
```

Comment:
RFC: Maybe we need to update immediately here?

Should happen in two cases:
- we don't have a list yet, so we already fetching it, (then request will be cancelled);
- a new participant (guest) joined and not in the list yet (would be a disaster for webinars, where new people are joining every time)

Reply:
Kept as-is, so will be updated in 3-15s
- [ ] add trigger to cancel this request (if everyone was found in very first server response, but data wasn't there yet)

Reply:
I agree to keep it as-is.

As we know from signaling message what attendees are joining, what about create a temporary participant with the info available and data will be completed once fetched ?

Reply:
we don't have enough information to show a participant from signaling until fetch. And we got all the actual info from it, there's no need to complete it later.
Though we can consider cache participants list 🤔